### PR TITLE
Manage error card and retry 

### LINKED ISSUE #25: Improve error handling in the front
Currently if the API returns an error, the UI keeps showing a loading state

### PR DESCRIPTION
**Pull request by zyhou**
fixes #25 

Capture

image

### CODE REVIEW

**Comment by Euregan** [comment]
#### 📝Changed routes:


| [**/**](https://aperoweb-git-manage-error-card.euregan.now.sh/) | [**/talks**](https://aperoweb-git-manage-error-card.euregan.now.sh/talks) |
|:-:|:-:|
|<a href="https://aperoweb-git-manage-error-card.euregan.now.sh/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=microlink.nyc3.digitaloceanspaces.com/bX6VrA0I8L95ytAMY3kb3oT_a8g_">(view full size)</a> |<a href="https://aperoweb-git-manage-error-card.euregan.now.sh/talks">Screenshot of /talks</a><br /><sup><a href="https://images.weserv.nl?url=microlink.nyc3.digitaloceanspaces.com/a_piW1YMIxNCSh7oM4yYM_EWfZY_">(view full size)</a> |







Commit bb431aafd3038024b245ff63f5142d1039e08d85 (https://aperoweb-git-manage-error-card.euregan.now.sh).

File: components/Card.js [hunk 1]
```diff
@@ -51,6 +51,32 @@ CardWithLoading.propTypes = {
     title: PropTypes.string,
 };
 
-CardWithLoading.defaultTypes = {
+CardWithLoading.defaultProps = {
     title: '',
 };
+
+export const CardWithError = ({ title, onClick }) => (
```
Comment (Euregan) [inline components/Card.js]: Shouldn't we change the name of the props to something indicating its use ? Like `onFailureRetry`

File: lib/useDataApi.js [hunk 2]
```diff
@@ -18,12 +18,20 @@ const dataFetchReducer = (state, action) => {
                 isLoading: false,
                 isError: true,
             };
+        case 'FETCH_RETRY':
+            return {
+                ...state,
+                isLoading: false,
```
Comment (Euregan) [inline lib/useDataApi.js]: Shouldn't that be loading when retrying?

File: pages/talks.js [hunk 3]
```diff
@@ -3,15 +3,15 @@ import React from 'react';
 import useDataApi from '../lib/useDataApi';
 import Layout from '../components/Layout';
 import Talk from '../components/Talk';
-import { CardWithLoading } from '../components/Card';
+import { CardWithLoading, CardWithError } from '../components/Card';
 import Grid from '../components/Grid';
 
-const LoadingCalendar = () =>
+const EmptyCalendar = ({ render }) =>
```
Comment (Euregan) [inline pages/talks.js]: `render` is the content?

Comment (zyhou) [reply pages/talks.js]: Yes, it's component. I will rename it `component`

File: lib/useDataApi.js (hunk 2 above)
Comment (zyhou) [reply lib/useDataApi.js]: Oh it's old code, I remove it

**Review by Euregan — APPROVED** [review]

**Pull request merged**